doc/May-2022-Upgrade.md

```diff
@@ -0,0 +1,397 @@
+    \q
+    ```
+    
+17. Prepare container to new version
+
+    New version of `kobo-docker` creates `kobotoolbox` database with PostGIS extension at first run.
+    To avoid trying to this at each subsequent start, a file is created with date of first run.
+    We need to add this file because extensions have been installed during this migration.
+
+    ```
+    echo $(date) > /var/lib/postgresql/data/kobo_first_run
+    echo "listen_addresses = '*'" >> /var/lib/postgresql/data/postgresql.conf
+    echo "host    all             all             10.0.0.0/8            trust" >> /var/lib/postgresql/data/pg_hba.conf
+    echo "host    all             all             172.0.0.0/8            trust" >> /var/lib/postgresql/data/pg_hba.conf
+    echo "host    all             all             192.0.0.0/8            trust" >> /var/lib/postgresql/data/pg_hba.conf
+    ```
+
+### MongoDB
+
+**Upgrading Mongo is easy and only implies a couple of stop/start.**
+
+1. Upgrade to WiredEngine 
+    Link to HackMD 
+
+2. Upgrade to 3.6
+
+    1. Stop `mongo` container
+
+        ```shell
+        user@computer:kobo-install$ python run.py --cb stop mongo  
+        ```
+       
+    2. Edit composer file `docker-compose.primary.backend.template.yml`
+
+    - Change image to `mongo:3.6`
+
+    ```
+    mongo:
+      image: mongo:3.6
+    ```
+   
+    Then start the container: `docker-compose up --force-recreate mongo`
+
+    Wait for MongoDB to be ready. You should see in the console the output below: 
+
+    ```
+    mongo_1        | {
+    mongo_1        | 	"numIndexesBefore" : 3,
+    mongo_1        | 	"numIndexesAfter" : 3,
+    mongo_1        | 	"note" : "all indexes already exist",
+    mongo_1        | 	"ok" : 1
+    mongo_1        | }
+    ```
+
+    From another terminal, enter the container and update compatibility version.
+
+    ```shell
+    root@mongo:/# mongo -u "$MONGO_INITDB_ROOT_USERNAME" -p "$MONGO_INITDB_ROOT_PASSWORD" admin
+    > db.adminCommand( { setFeatureCompatibilityVersion: "3.6" } )
+    { "ok" : 1 }
+    > exit
+    bye
+    root@mongo:/# exit
+    ```
+3. Upgrade to 4.0, 4.2, 4.4 and 5.0
+
+    Repeat step above for each version and replace the version accordingly.
+    You **must** upgrade each version one by one.
+    
+    Then start the container: `docker-compose up --force-recreate mongo`
+
+    Done!
+
+
+## Tests
+
+1. Test if upgrade is successful
+
+    Start your containers as usual.
+
+    ```shell
+    user@computer:kobo-install$ python run.py  
+    ```
+
+    Log into one of your user accounts and validate everything is working as expected.         
+
+2. Clean up
+
+   If everything is ok, you can now delete data from `PostgreSQL 9.5`
+
+   1. Stop containers
+    
+       ```shell
+       user@computer:kobo-install$ python run.py --stop  
+       ```
+    
+   2. Rename folder
+    
+        ```shell
+        user@computer:kobo-docker$ sudo rm -rf .vols/db
+        user@computer:kobo-docker$ sudo mv .vols/db14 .vols/db
```

Comment:
We should probably add that the docker-compose file should also be updated with `- ./.vols/db:/var/lib/postgresql/data`

Reply:
Indeed.